shard/committee/assignment.go

```diff
@@ -151,107 +222,32 @@ func eposStakedCommittee(
 		}
 	}
 
-	if stakedSlotsCount == 0 {
+	completedEPoSRound, err := NewEPoSRound(stakerReader)
+
+	if completedEPoSRound.MaximumExternalSlotAllowed == 0 {
```

Comment:
MaximumExternalSlotAllowed is never set to 0 even if there is no external validators elected.